### ISSUE TITLE
Support deleting multiple deployments 

### ISSUE THREAD

**Issue by khuyentran1401** [PrefectHQ/prefect#7363]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Right now, it is time-consuming to delete multiple deployments at once. For every deployment I want to delete, I need to click a three-dot button -> click Delete.

This is not ideal when there are multiple deployments that users want to delete at once.

![Screen Shot 2022-10-28 at 10 40 04 AM](https://user-images.githubusercontent.com/49108771/198682084-a3b4577e-e95e-44f4-81d4-3fb8e0f1b7a4.png)

### Describe the proposed behavior

It would be great to add the ability to delete multiple deployments like how we did with the Flow Runs page.

![Oct-28-2022 11-04-37](https://user-images.githubusercontent.com/49108771/198682623-d134b1a5-41d7-473b-a42d-a06af0d7bd87.gif)

**Comment by discdiver** [comment]
Applies to deleting many objects in the UI. A checkbox to select all would be great.

**Comment by zhen0** [comment]
complement to #6025

**Comment by zhen0** [comment]
@billpalombi @brigidgliwa - could we add some design for this?  One to consider alongside the tags visibility question in the deployments table.

**Comment by brigidgliwa** [comment]
Yes I will add a design asap

**Comment by brigidgliwa** [comment]
Here is the link to the figma designs https://www.figma.com/file/WQwcuqBAxug8JqLigWaztu/Designs?node-id=2119%3A24235

<img width="1326" alt="01-Deployments Delete" src="https://user-images.githubusercontent.com/104096908/199549711-9909513f-7afa-4c3b-9ab3-c095e91ca830.png">
<img width="1326" alt="02-Deployments Delete" src="https://user-images.githubusercontent.com/104096908/199549713-c620b1be-8068-4d0f-9900-aff3f74c5347.png">

**Comment by brigidgliwa** [comment]
Here is the link to the figma designs: https://www.figma.com/file/WQwcuqBAxug8JqLigWaztu/Designs?node-id=2119%3A24809

<img width="1326" alt="01-Deployments Delete" src="https://user-images.githubusercontent.com/104096908/199550070-a3a5d115-ee8e-417b-959c-d52797ff7fc7.png">
<img width="1326" alt="02-Deployments Delete" src="https://user-images.githubusercontent.com/104096908/199550071-2a1dac04-7666-4b6e-aad5-1bf2cf73d992.png">

**Comment by billpalombi** [comment]
Do I assume correctly that, if deployments are paginated, and the user selects the "All" checkbox, only the deployments visibly displayed on the platform are selected?

**Comment by brigidgliwa** [comment]
Yes I would assume it works similar to Gmail that you are only selecting the ones on the page you are on...if you go from page 1 to page 2 the check all box would reset, but if you went back to page 1 they should all still be selected

**Comment by billpalombi** [comment]
Understood, that is fine for now, just making sure.

**Comment by zhen0** [comment]
@brigidgliwa - Is this ready to move to development?

**Comment by brigidgliwa** [comment]
I updated the design to show the number of selected items and the actions right next to it (very similar to how GitHub does it). Search and filters will still be shown but sort will not.
<img width="1326" alt="02-Deployments Delete" src="https://user-images.githubusercontent.com/104096908/203437370-c152e9ee-28d6-4980-8103-56a90c86c61e.png">

**Comment by billpalombi** [comment]
Awesome, thank you

**Comment by zhen0** [comment]
Thanks @brigidgliwa! 

Since the action here is mainly in the table I'll flag that #6025 is a very similar piece of work and it might make sense to do them together.
